server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1317,6 +1317,7 @@ public <T extends RelationMetadata> T getRelation(RelationName relation) {
      * {@code null} values returned from {@code as} are excluded from the result.
      * This can be used to filter based on state or similar.
      * </p>
+     * @param partitionValues filter by a single partition. Use `List.of()` to include all partitions.
```

Comment:
Is it worth adding a sugar method that doesn't include this parameter?  We pass `List.of()` in every call-site bar one I think?